### DISCUSSION TITLE
Is there a plan to support Django ASGI to workers packages?

### DISCUSSION THREAD

**Discussion by epuerta9** [cloudflare/workerd#3032] [Python Packages]
Perusing through the cloudflare docs and python workers support with packages is really appealing. I see fastapi is supported and I'm wondering if Django would ever make that list. My project is pretty dependent on Django and if ASGI is the blocker, django, django-ninja, and other packages have native async support. I saw last discussion around Django was April

**Comment by hoodmane** [comment]
Yeah we'd even like to support Django synchronous. Unfortunately, it's not clear how long it'll take us to get around to working on this.

**Comment by ryanking13** [comment]
django now works in Python workers. You can choose either WSGI or ASGI (https://developers.cloudflare.com/workers/languages/python/packages/django/). Note that Django ASGI is not a real ASGI and relies on background threading. Since Python workers is single threaded, all the operations happen in the same thread.
